src/DistributedComputations/distributed_fft_based_poisson_solver.jl

```diff
@@ -55,155 +47,159 @@ Algorithm for two-dimensional decompositions
 For two-dimensional decompositions (useful for three-dimensional problems),
 there are three forward transforms, three backward transforms,
 and four transpositions requiring MPI communication. In the schematic below, the first
-dimension is always the local dimension. In our implementation of the PencilFFTs algorithm,
-we require _either_ `Nz >= Rx`, _or_ `Nz >= Ry`, where `Nz` is the number of vertical cells,
+dimension is always the local dimension. In our implementation we require
+`Nz ≥ Ry` and `Nx ≥ Ry` with the additional constraint that `Nz % Ry = 0` and `Ny % Rx = 0`.
 `Rx` is the number of ranks in ``x``, and `Ry` is the number of ranks in ``y``.
-Below, we outline the algorithm for the case `Nz >= Rx`.
-If `Nz < Rx`, but `Nz > Ry`, a similar algorithm applies with ``x`` and ``y`` swapped:
 
-1. `first(storage)` is initialized with layout ``(z, x, y)``.
+1. `storage.zfield`, partitioned over ``(x, y)`` is initialized with the `rhs`.
 2. Transform along ``z``.
-3  Transpose + communicate to `storage[2]` in layout ``(x, z, y)``,
-   which is distributed into `(Rx, Ry)` processes in ``(z, y)``.
-4. Transform along ``x``.
-5. Transpose + communicate to `last(storage)` in layout ``(y, x, z)``,
-   which is distributed into `(Rx, Ry)` processes in ``(x, z)``.
-6. Transform in ``y``.
+3  Transpose + communicate to `storage.yfield` partitioned into `(Rx, Ry)` processes in ``(x, z)``.
+4. Transform along ``y``.
+5. Transpose + communicate to `storage.xfield` partitioned into `(Rx, Ry)` processes in ``(y, z)``.
+6. Transform in ``x``.
```

Comment:
It's not clear to me what `storage.zfield` (and x and y) means. Granted, I'm not familiar with the FFT/poisson algorithms, so maybe it's obvious for other developers.

The rest of this part makes sense to me though.

Reply:
Sounds same question as mine..